src/PonyVCO.cpp

```diff
@@ -0,0 +1,405 @@
+			double pwp = p + 2 * pw;			// phase after pw (pw in [0, 1])
+			pwp += (pwp > 1) * -2;     			// modulo on [-1, +1]
+			sawOffsetBuff[i] = (pwp * pwp * pwp - pwp) / 6.0;	// eq 11
+		}
+		return (sawOffsetBuff[0] - 2.0 * sawOffsetBuff[1] + sawOffsetBuff[2]);
+	}
+
+	float wavefolder(float x, float xt) {
+		return stage2.process(stage1.process(x, xt));
+	}
+
+	json_t* dataToJson() override {
+		json_t* rootJ = json_object();
+		json_object_set_new(rootJ, "blockTZFMDC", json_boolean(blockTZFMDC));
+		json_object_set_new(rootJ, "removePulseDC", json_boolean(removePulseDC));
+		json_object_set_new(rootJ, "oversamplingIndex", json_integer(oversampler.getOversamplingIndex()));
+		return rootJ;
+	}
+
+	void dataFromJson(json_t* rootJ) override {
+
+		json_t* blockTZFMDCJ = json_object_get(rootJ, "blockTZFMDC");
+		if (blockTZFMDCJ) {
+			blockTZFMDC = json_boolean_value(blockTZFMDCJ);
+		}
+
+		json_t* removePulseDCJ = json_object_get(rootJ, "removePulseDC");
+		if (removePulseDCJ) {
+			removePulseDC = json_boolean_value(removePulseDCJ);
+		}
+
+		json_t* oversamplingIndexJ = json_object_get(rootJ, "oversamplingIndex");
+		if (oversamplingIndexJ) {
+			oversamplingIndex = json_integer_value(oversamplingIndexJ);
+			onSampleRateChange();
+		}
+	}
+};
+
+
+struct PonyVCOWidget : ModuleWidget {
+	PonyVCOWidget(PonyVCO* module) {
+		setModule(module);
+		setPanel(createPanel(asset::plugin(pluginInstance, "res/panels/PonyVCO.svg")));
+
+		addChild(createWidget<Knurlie>(Vec(RACK_GRID_WIDTH, 0)));
+		addChild(createWidget<Knurlie>(Vec(RACK_GRID_WIDTH, RACK_GRID_HEIGHT - RACK_GRID_WIDTH)));
+
+		addParam(createParamCentered<Davies1900hDarkGreyKnob>(mm2px(Vec(10.0, 14.999)), module, PonyVCO::FREQ_PARAM));
+		addParam(createParam<CKSSHoriz4>(mm2px(Vec(5.498, 27.414)), module, PonyVCO::RANGE_PARAM));
+		addParam(createParam<BefacoSlidePotSmall>(mm2px(Vec(12.65, 37.0)), module, PonyVCO::TIMBRE_PARAM));
+		addParam(createParam<CKSSVert7>(mm2px(Vec(3.8, 40.54)), module, PonyVCO::OCT_PARAM));
+		addParam(createParam<CKSSHoriz4>(mm2px(Vec(5.681, 74.436)), module, PonyVCO::WAVE_PARAM));
+
+		addInput(createInputCentered<BefacoInputPort>(mm2px(Vec(5.014, 87.455)), module, PonyVCO::TZFM_INPUT));
+		addInput(createInputCentered<BefacoInputPort>(mm2px(Vec(14.978, 87.455)), module, PonyVCO::TIMBRE_INPUT));
+		addInput(createInputCentered<BefacoInputPort>(mm2px(Vec(5.014, 100.413)), module, PonyVCO::VOCT_INPUT));
+		addInput(createInputCentered<BefacoInputPort>(mm2px(Vec(14.978, 100.413)), module, PonyVCO::SYNC_INPUT));
+		addInput(createInputCentered<BefacoInputPort>(mm2px(Vec(5.014, 113.409)), module, PonyVCO::VCA_INPUT));
+
+		addOutput(createOutputCentered<BefacoOutputPort>(mm2px(Vec(15.0, 113.363)), module, PonyVCO::OUT_OUTPUT));
+	}
+
+	void appendContextMenu(Menu* menu) override {
+		PonyVCO* module = dynamic_cast<PonyVCO*>(this->module);
+		assert(module);
+
+		menu->addChild(new MenuSeparator());
+		menu->addChild(createSubmenuItem("Hardware compatibility", "",
```

Comment:
Befaco have explicity asked that the VCV module _technically_ can behave as the real one, even if the defaults are not the same. Also any differences with hardware are noted in the README